number/number.go

```diff
@@ -0,0 +1,156 @@
+package number
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+
+	messagebird "github.com/messagebird/go-rest-api"
+)
+
+const (
+	apiRoot              = "https://numbers.messagebird.com/v1"
+	pathNumbers          = "phone-numbers"
+	pathNumbersAvailable = "available-phone-numbers"
+)
+
+type Number struct {
+	Number   string
+	Country  string
+	Region   string
+	Locality string
+	Features []string
+	Tags     []string
+	Type     string
+	Status   string
+}
+
+type NumberList struct {
+	Offset     int
+	Limit      int
+	Count      int
+	TotalCount int
+	Items      []*Number
+}
+
+type NumberSearchingList struct {
+	Items []*Number
+	Limit int
+	Count int
+}
+
+type NumberListParams struct {
+	Limit         int
+	Offset        int
+	Number        string
+	Country       string
+	Region        string
+	Locality      string
+	Features      []string
+	Type          []string
+	Status        string
+	SearchPattern string
+}
+
+// request does the exact same thing as Client.Request. It does, however,
+// prefix the path with the Numbers API's root. This ensures the client
+// doesn't "handle" this for us: by default, it uses the REST API.
+func request(c *messagebird.Client, v interface{}, method, path string, data interface{}) error {
+	fmt.Println(path)
+	return c.Request(v, method, fmt.Sprintf("%s/%s", apiRoot, path), data)
+}
+
+// List get all purchased phone numbers
+func List(c *messagebird.Client, listParams *NumberListParams) (*NumberList, error) {
+	uri := GetPath(listParams, pathNumbers)
+
+	numberList := &NumberList{}
+	if err := request(c, numberList, http.MethodGet, uri, nil); err != nil {
+		return nil, err
+	}
+	return numberList, nil
+}
+
+// Search for phone numbers available for purchase
+func Search(c *messagebird.Client, cc string, listParams *NumberListParams) (*NumberSearchingList, error) {
+	uri := GetPath(listParams, pathNumbersAvailable+"/"+cc)
+
+	numberList := &NumberSearchingList{}
+	if err := request(c, numberList, http.MethodGet, uri, nil); err != nil {
+		return nil, err
+	}
+
+	return numberList, nil
+}
+
+// Read get a purchased phone number
+func Read(c *messagebird.Client, phoneNumber string) (*Number, error) {
+	if len(phoneNumber) < 5 {
+		return nil, fmt.Errorf("a phoneNumber is too short")
+	}
+
+	uri := fmt.Sprintf("%s/%s", pathNumbers, phoneNumber)
+
+	number := &Number{}
+	if err := request(c, number, http.MethodGet, uri, nil); err != nil {
+		return nil, err
+	}
+
+	return number, nil
+}
+
+// GetPath get the full path for the request
+func GetPath(listParams *NumberListParams, path string) string {
```

Comment:
Why is this  method exported ?

Reply:
Looks like the function name is a bit misleading as well.  IMHO ,  use `paramsForMessageList` and  `params.Encode()`  as such instead of `GetPath`